File: src/main/java/org/taktik/connector/business/mycarenet/mhm/service/MhmService.java
package org.taktik.connector.business.mycarenet.mhm.service;

import be.fgov.ehealth.mycarenet.attest.protocol.v2.CancelAttestationResponse;
import be.fgov.ehealth.mycarenet.attest.protocol.v2.SendAttestationResponse;
import be.fgov.ehealth.mycarenet.commons.protocol.v3.SendRequestType;
import be.fgov.ehealth.mycarenet.mhm.protocol.v1.SendSubscriptionResponse;
import org.taktik.connector.technical.exception.TechnicalConnectorException;
import org.taktik.connector.technical.service.sts.security.SAMLToken;

public interface MhmService {
   SendSubscriptionResponse sendSubscription(SAMLToken var1, SendRequestType var2) throws TechnicalConnectorException;
}

File: src/main/kotlin/org/taktik/connector/business/mhm/MhmService.kt
package org.taktik.connector.business.mhm

import be.fgov.ehealth.mycarenet.mhm.protocol.v1.CancelSubscriptionRequest
import be.fgov.ehealth.mycarenet.mhm.protocol.v1.CancelSubscriptionResponse
import be.fgov.ehealth.mycarenet.mhm.protocol.v1.NotifySubscriptionClosureRequest
import be.fgov.ehealth.mycarenet.mhm.protocol.v1.NotifySubscriptionClosureResponse
import be.fgov.ehealth.mycarenet.mhm.protocol.v1.SendSubscriptionRequest
import be.fgov.ehealth.mycarenet.mhm.protocol.v1.SendSubscriptionResponse
import org.taktik.connector.technical.service.sts.security.SAMLToken
import org.taktik.freehealth.middleware.dto.mhm.CancelSubscriptionResult
import org.taktik.freehealth.middleware.dto.mhm.EndSubscriptionResult

interface MhmService {
    fun sendSubscription(token: SAMLToken, request: SendSubscriptionRequest, soapAction: String): SendSubscriptionResponse
    fun cancelSubscription(token: SAMLToken, request: CancelSubscriptionRequest, soapAction: String) : CancelSubscriptionResponse
    fun notifySubscriptionClosure(token: SAMLToken, request: NotifySubscriptionClosureRequest, soapAction: String) : NotifySubscriptionClosureResponse
}
